acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/tls/ClientSideTlsAcceptanceTest.java

```diff
@@ -186,29 +186,55 @@ void ethSignerDoesNotConnectToServerNotSpecifiedInTrustStore(@TempDir Path workD
   @Test
   void missingKeyStoreForEthSignerResultsInEthSignerTerminating(@TempDir Path workDir)
       throws Exception {
-    final TlsCertificateDefinition serverPresentedCert =
-        TlsCertificateDefinition.loadFromResource("tls/cert1.pfx", "password");
-    final TlsCertificateDefinition ethSignerCert =
+    final TlsCertificateDefinition missingServerCert =
         new TlsCertificateDefinition(
             workDir.resolve("Missing_keyStore").toFile(), "arbitraryPassword");
+    final TlsCertificateDefinition serverCert =
+        TlsCertificateDefinition.loadFromResource("tls/cert1.pfx", "password");
+    final TlsCertificateDefinition ethSignerCert =
+        TlsCertificateDefinition.loadFromResource("tls/cert2.pfx", "password2");
+
+    final HttpServer web3ProviderHttpServer =
+        serverFactory.create(serverCert, ethSignerCert, workDir);
 
-    // Ports are arbitrary as EthSigner should exit
-    signer = createSigner(ethSignerCert, serverPresentedCert, 9000, 9001, workDir);
+    signer =
+        createSigner(
+            missingServerCert, ethSignerCert, web3ProviderHttpServer.actualPort(), 0, workDir);
     signer.start();
-    waitFor(() -> assertThat(signer.isRunning()).isFalse());
+
+    // the actual connection to downstream server should fail as an internal error (500) since the
+    // keystore is invalid ...
+    assertThatThrownBy(() -> signer.accounts().balance("0x123456"))
+        .isInstanceOf(ClientConnectionException.class)
+        .hasMessageContaining(String.valueOf(INTERNAL_SERVER_ERROR.code()));
   }
 
   @Test
   void incorrectPasswordForDownstreamKeyStoreResultsInEthSignerTerminating(@TempDir Path workDir)
       throws Exception {
-    final TlsCertificateDefinition serverPresentedCert =
+    final TlsCertificateDefinition serverPresentedCertWithInvalidPassword =
+        TlsCertificateDefinition.loadFromResource("tls/cert1.pfx", "wrong_password");
+    final TlsCertificateDefinition serverCert =
         TlsCertificateDefinition.loadFromResource("tls/cert1.pfx", "password");
     final TlsCertificateDefinition ethSignerCert =
-        TlsCertificateDefinition.loadFromResource("tls/cert1.pfx", "wrong_password");
+        TlsCertificateDefinition.loadFromResource("tls/cert2.pfx", "password2");
```

Comment:
Is "password2" the wrong password for this keystore? If so I think "wrong_password" makes this more obvious.

Reply:
ah I see wrong_password is used a couple of lines above, missed that